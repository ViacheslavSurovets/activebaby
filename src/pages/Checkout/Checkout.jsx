import React from 'react';
import { connect } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { createStructuredSelector } from 'reselect';
import { CheckoutItem } from '@components';
import { selectCartItems, selectCartTotal } from '@redux/cart/cart.selectors';
import { StripeButton } from '@components';
import {
  CheckoutHeader,
  CheckoutHeaderBlock,
  CheckoutPage,
  HeaderBlockText,
  CheckoutTotal
} from './styles';
import PropTypes from 'prop-types';


const Checkout = ( { cartItems, total } ) => {
  const { t } = useTranslation ();

  return (
    <CheckoutPage>
      <CheckoutHeader>
        <CheckoutHeaderBlock>
          <HeaderBlockText>{ t ( 'checkoutPage.product' ) }</HeaderBlockText>
        </CheckoutHeaderBlock>
        <CheckoutHeaderBlock>
          <HeaderBlockText>{ t ( 'checkoutPage.description' ) }</HeaderBlockText>
        </CheckoutHeaderBlock>
        <CheckoutHeaderBlock>
          <HeaderBlockText>{ t ( 'checkoutPage.quantity' ) }</HeaderBlockText>
        </CheckoutHeaderBlock>
        <CheckoutHeaderBlock>
          <HeaderBlockText>{ t ( 'checkoutPage.price' ) }</HeaderBlockText>
        </CheckoutHeaderBlock>
        <CheckoutHeaderBlock>
          <HeaderBlockText>{ t ( 'checkoutPage.remove' ) }</HeaderBlockText>
        </CheckoutHeaderBlock>
      </CheckoutHeader>
      {
        cartItems.map ( cartItem =>
          <CheckoutItem key={ cartItem.id } cartItem={ cartItem } />
        )
      }
      <CheckoutTotal>{ total }BYN</CheckoutTotal>
      <StripeButton price={ total }/>
    </CheckoutPage>


  );
};

const mapStateToProps = createStructuredSelector ( {
  cartItems: selectCartItems,
  total: selectCartTotal
} );

export default connect ( mapStateToProps ) ( Checkout );


Checkout.propTypes = {
  cartItems: PropTypes.array,
  total: PropTypes.number
};
