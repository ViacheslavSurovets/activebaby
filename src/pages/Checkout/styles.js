import styled from 'styled-components';
import { commonStyleTextTwoRem } from '@src/styles/commonStylesAndStyledComponents';
import { mobile } from '@media';

export const CheckoutPage = styled.section`
  width: 55%;
  min-height: 90vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 5rem auto 0;
  user-select: none;
  ${ mobile }{
    overflow: auto;
    min-width: 70rem;
  }
`;

export const CheckoutHeader = styled.header`
  display: flex;
  justify-content: space-between;
  width: 100%;
  padding: 1rem 0;
  border-bottom: 1px solid ${ ( { theme } ) => theme.darkGray };
`;

export const CheckoutHeaderBlock = styled.div`
  text-transform: capitalize;
  width: 23%;

  &:last-child {
     width: 8%;
  }
`;


export const CheckoutTotal = styled.div`
 margin-top: 3rem;
 margin-left: auto;
 font-size: 3.6rem;
`;

export const HeaderBlockText = styled.span`
  ${ commonStyleTextTwoRem };
`;
