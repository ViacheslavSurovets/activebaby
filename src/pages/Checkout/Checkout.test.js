import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Checkout from './Checkout';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('Checkout', ()=> {
  it('Checkout Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Checkout  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});
