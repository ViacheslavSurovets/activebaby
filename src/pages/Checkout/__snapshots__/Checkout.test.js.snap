// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout Checkout Snapshot 1`] = `ShallowWrapper {}`;
